check-rabbitmq-exchange/lib/Nagios/Check/RabbitMQ/Process.hs

```diff
@@ -8,13 +8,13 @@ import           Data.Aeson
 import           Data.ByteString.Lazy.Char8  (ByteString)
 import qualified Data.ByteString.Lazy.Char8  as BSL
 import           Data.Maybe
-
+import qualified Data.Text as T
 import           Nagios.Check.RabbitMQ.Types
 import           System.Nagios.Plugin
 
 checkRawExchange :: ByteString -> CheckOptions -> IO ()
 checkRawExchange bs opts = case eitherDecode bs of
-    Left  e -> putStrLn $ "Decoding failed with: " ++ e
+    Left  e -> runNagiosPlugin $ addResult Warning $ T.pack ( "Decode failed with: " ++ e )
```

Comment:
I think this should be `UNKNOWN`, not `WARNING` - this is a rabbit exchange check, and failing to decode the API output means we don't know the state of the exchange.